lib/ruby_lsp/ruby_document.rb

```diff
@@ -257,6 +257,37 @@ def locate_node(position, node_types: [])
       )
     end
 
+    sig { returns(String) }
+    def test_library
```

Comment:
Part of the advantage of looking at the ancestors is that we become more accurate even in very uncommon setups, like mixing test frameworks - which is something that came up a few times.

Considering that test files can have multiple test classes and they can inherit from different parents, is this approach flexible enough to account for that?

What was the idea behind putting it in document versus doing it in the code lens implementation?